crates/flox-rust-sdk/src/models/environment/generations.rs

```diff
@@ -0,0 +1,328 @@
+#![allow(dead_code)] //
+use std::collections::BTreeMap;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use chrono::{DateTime, Utc};
+use flox_types::version::Version;
+use serde::{Deserialize, Serialize};
+use serde_with::{DeserializeFromStr, SerializeDisplay};
+use thiserror::Error;
+
+use super::path_environment::{Original, PathEnvironment};
+use super::{PathPointer, ENV_DIR_NAME, LOCKFILE_FILENAME, MANIFEST_FILENAME};
+use crate::providers::git::{GitCommandProvider, GitProvider};
+
+const GENERATIONS_METADATA_FILE: &str = "metadata.json";
+
+/// A representation of the generations of an environment
```

Comment:
Suggestion: It would be helpful if there was some inline documentation here describing the layout of the realized generations directory.

Reply:
added